src/Serilog.Sinks.AzureTableStorage/Sinks/AzureTableStorageWithProperties/AzureTableStorageEntityFactory.cs

```diff
@@ -61,9 +61,15 @@ public static DynamicTableEntity CreateEntityWithProperties(LogEvent logEvent, I
 
             List<KeyValuePair<ScalarValue, LogEventPropertyValue>> additionalData = null;
             var count = dynamicProperties.Count;
+			bool isNumeric;
+			int parsedNumber;
+
             foreach (var logProperty in logEvent.Properties)
             {
-                if (count++ < _maxNumberOfPropertiesPerRow - 1)
+				isNumeric = int.TryParse(logProperty.Key, out parsedNumber);
```

Comment:
Considering this use case a bit more, it may be useful to treat numeric properties separately from generally invalid property names. If someone uses `{0}`, `{1}`, etc. then they're obviously not wanting these to be stored as a separate property in the table (otherwise they'd have specified a name) and likely not in the `AggregateProperties` table property either, as long as they are in the rendered message. If it's not a numeric property name, then a separate check to ensure it conforms to C# identifier naming conventions makes sense. I'll add that function and validate the name using a C# identifier regex.

In short, numeric properties should probably be skipped. All other properties are either stored in a table property if the name is valid or, if the name isn't valid, in the `AggregateProperties` table property.